.github/workflows/python-package.yml

```
@@ -0,0 +1,52 @@
+# This workflow will install Python dependencies, run tests and lint with a variety of Python versions
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: Python package
+
+on: pull_request
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.12"]
+        poetry-version: ["1.8.3"]
+
+    permissions:
+      pull-requests: write
+    
+    steps:
+    - uses: actions/checkout@v4
+    
+    - name: "Setup Python, Poetry and Dependencies"
```

```suggestion
    - name: Setup Python, Poetry and Dependencies
```
Can skip quotation marks here to be consistent with other sections